tsdb/store.go

```diff
@@ -305,10 +305,6 @@ func (s *Store) WriteToShard(shardID uint64, points []Point) error {
 
 func (s *Store) CreateMapper(shardID uint64, stmt influxql.Statement, chunkSize int) (Mapper, error) {
 	shard := s.Shard(shardID)
-	if shard == nil {
```

Comment:
This issue is that `CreateMapper` returns `nil` which causes a panic in here: https://github.com/influxdb/influxdb/pull/3906/files#diff-06abda922c6e558f8a9fe56e04b7ed5eL94